### PR TITLE
Use stackalloc in NetHelpers

### PR DESCRIPTION
**Pull request by ilyfairy**

### CODE REVIEW

**Comment by xPaw** [comment]
Can NetworkToHostOrder/HostToNetworkOrder be used?

File: SteamKit2/SteamKit2/Util/NetHelpers.cs [hunk 1]
```diff
@@ -9,36 +9,37 @@
...
 
         public static IPAddress GetIPAddress( uint ipAddr )
         {
-            byte[] addrBytes = BitConverter.GetBytes( ipAddr );
-            Array.Reverse( addrBytes );
-
+            Span<byte> addrBytes = stackalloc byte[ 4 ];
+            BitConverter.TryWriteBytes( addrBytes, IPAddress.NetworkToHostOrder( Unsafe.BitCast<uint, int>( ipAddr ) ) );
+            
             return new IPAddress( addrBytes );
```
Comment (xPaw) [inline SteamKit2/SteamKit2/Util/NetHelpers.cs:34]: can't you construct from the int now without the byte array?

Comment (ilyfairy) [reply SteamKit2/SteamKit2/Util/NetHelpers.cs]: has been changed to new IPAddress(long address)

**Comment by yaakov-h** [comment]
Do we know if NetworkToHostOrder is correct here when running on arm64?

**Comment by ilyfairy** [comment]
> Do we know if NetworkToHostOrder is correct here when running on arm64?


image

**Comment by yaakov-h** [comment]
Do the Steam servers get the same value when they receive and parse the `CMsgIPAddress` though?

I care more about that host than the one running SK2.

**Comment by xPaw** [comment]
We have tests for this and one of the CI runners is macos-14 arm64 and it passes, so probably fine.

**Comment by xPaw** [comment]
Well this broke something:

> Unexpected exception propogated back to NetLoop: System.ArgumentOutOfRangeException: newAddress ('1844xxxxxxxxxxxxxx') must be less than or equal to '4294967295'. (Parameter 'newAddress')

at SteamKit2.NetHelpers.GetIPAddress(UInt32 ipAddr) in NetHelpers.cs:line 31

**Pull request merged**